lib/chef/provider/ifconfig/systemd.rb

```diff
@@ -0,0 +1,174 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require "ipaddr"
+require "iniparse"
+
+require "chef/mixin/which"
+require "chef/mixin/shell_out"
+
+class Chef
+  class Provider
+    class Ifconfig
+      class Systemd < Chef::Provider::Ifconfig
+        extend Chef::Mixin::Which
+        include Chef::Mixin::ShellOut
+        extend Chef::Mixin::ShellOut
+
+        provides :ifconfig do
+          systemd_unit_enabled?("systemd-networkd")
+        end
+
+        def load_current_resource
+          @current_resource = Chef::Resource::Ifconfig.new(new_resource.name)
+
+          @status = shell_out_with_systems_locale!("ip", "-json", "addr", "list")
+          json = JSON.parse @status.stdout
+          interfaces = json.each_with_object({}) { |iface, acc| acc[iface["ifname"]] = iface }
+
+          iface = driver_name
+          interface = interfaces.fetch(iface, nil)
+          if !interface.nil?
+            current_resource.target(new_resource.target)
+            current_resource.device(new_resource.device)
+            current_resource.hwaddr(interface["address"])
+            current_resource.mtu(interface["mtu"])
+            address = interface["addr_info"].select { |i| i["label"] == new_resource.device && i["family"] == new_resource.family }.first
+            return current_resource if address.nil?
+            netmask = get_netmask(address["prefixlen"])
+            current_resource.inet_addr(address["local"])
+            current_resource.bcast(address["broadcast"])
+            current_resource.mask(netmask)
+          end
+          current_resource
+        end
+
+        def action_add
+          unit = generate_network_unit
+          converge_by "Creating network unit: #{unit_name}" do
+            declare_resource(:directory, unit_dir) do
+              mode "0755"
+              owner "root"
+              group "root"
+            end
+
+            declare_resource(:file, ::File.join(unit_dir, unit_name)) do
+              content unit
+              owner "root"
+              group "root"
+              mode "0644"
+              # It appears that systemd-analyze doesn't currently support network units
+              # verify :systemd_unit
+            end
+          end
+        end
+
+        def self.systemd_unit_enabled?(unit)
+          systemctl_execute!("is-enabled", unit).exitstatus == 0
+        end
+
+        def self.systemctl_execute!(action, unit)
+          systemctl_path = which("systemctl")
+          systemctl_cmd = "#{systemctl_path} --system"
+          shell_out_with_systems_locale!("#{systemctl_cmd} #{action} #{Shellwords.escape(unit)}")
```

Comment:
these should just be `shell_out(..., default_env: false)` now.

Reply:
actually i think none of this should be `default_env: false` or `_with_systems_locale` and they should just be straight `shell_out`